src/main/java/Denim/Storage/TaskIO.java

```diff
@@ -0,0 +1,275 @@
+package denim.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import denim.TaskList;
+import denim.exceptions.DenimException;
+import denim.tasks.Deadline;
+import denim.tasks.Event;
+import denim.tasks.Task;
+import denim.tasks.Todo;
+
+/**
+ * Handles the input and output operations for task data in the Denim application.
+ * This class is responsible for reading tasks from and writing tasks to a file,
+ * managing file creation, and processing task data.
+ */
+public class TaskIo {
+
+    /**
+     * Represents the status of the file or directory during task I/O operations.
+     */
+    enum FileStatus {
+        DIRECTORY_DOES_NOT_EXIST,
+        FILE_DOES_NOT_EXIST
+    }
+
+    static final String TODO = "T";
+    static final String EVENT = "E";
+    static final String DEADLINE = "D";
+
+    private final File taskFile;
+
+    /**
+     * Constructs a TaskIO object associated with the specified file path.
+     *
+     * @param pathname The path to the file where tasks are stored.
+     */
+    public TaskIo(String pathname) {
+        taskFile = new File(pathname);
+    }
+
+    /**
+     * Creates a save point by handling directory or file creation based on the provided status.
+     *
+     * @param status The status indicating whether the directory or file does not exist.
+     * @param sc     The scanner to receive user input for creating the necessary files.
+     * @throws DenimException If an error occurs during file or directory creation.
+     */
+    private void createSavePoint(FileStatus status, Scanner sc) throws DenimException {
+        switch (status) {
+        case DIRECTORY_DOES_NOT_EXIST:
```

Comment:
It's interesting that you chose to use an enum for file status. While it's definitely good practice, I'm curious about the point at which enums might become less advantageous compared to using else if statements.
Would like to know your opinion

Reply:
Hi @kimxw , the reason why I used `enums` is just for better readability than anything else. I think it is fine to use 
`if else` statement here as there are only a few cases to consider here.